### PR TITLE
[DEV-5726] privileged default user

### PR DESCRIPTION
**Pull request by gaetano-guerriero**
This change is [Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3265)

### CODE REVIEW

File: server/athenian/api/auth.py [hunk 1]
```diff
@@ -687,23 +687,23 @@ async def wrapper(request: EspecificoRequest):
...
 
     wraps(wrapped_disable_default_user, func)
     return wrapped_disable_default_user
 
 
-def ensure_non_default_user(request: AthenianWebRequest):
-    """Raise exception if the user on the request is the default user."""
-    if request.is_default_user:
+def forbid_unprivileged_default_user(request: AthenianWebRequest):
+    """Raise exception if the request user is the default user and has not god privileges."""
```
Comment (vmarkovtsev) [inline server/athenian/api/auth.py:705]:
```suggestion
    """Raise exception if the request user is the default user and has no god privileges."""
```

File: server/athenian/api/auth.py [hunk 2]
```diff
@@ -687,23 +687,23 @@ async def wrapper(request: EspecificoRequest):
...
+    if request.is_default_user and getattr(request, "god_id", None) is None:
```
Comment (vmarkovtsev) [inline server/athenian/api/auth.py:706]:
```suggestion
    if request.is_default_user and not hasattr(request, "god_id"):
```
(shorter, the same as in internal/account.py `get_user_account_status_from_request`)

**Review by vmarkovtsev — APPROVED** [review]
Question: did we consider setting `--force-user` in e2e tests? This way, we should bypass Auth0 and all the requests will happen on behalf of that user.

**Comment by gaetano-guerriero** [comment]
The would be great, we could use another user, I will check if it plays well with demo mode.

**Pull request closed without merging**